cms/static/js/views/move_xblock_list.js

```diff
@@ -0,0 +1,170 @@
+                this.parentInfo.parent = this.childrenInfo.children[newParentIndex];
+            } else if (direction === 'backward') {
+                // new parent will be one of visitedAncestors
+                this.parentInfo.parent = this.visitedAncestors[newParentIndex];
+                // remove visited ancestors
+                this.visitedAncestors.splice(newParentIndex);
+            }
+
+            this.visitedAncestors.push(this.parentInfo.parent);
+
+            if (this.parentInfo.parent.get('child_info')) {
+                this.childrenInfo.children = this.parentInfo.parent.get('child_info').children;
+            } else {
+                this.childrenInfo.children = [];
+            }
+
+            this.setDisplayedXBlocksCategories();
+            this.render();
+        },
+
+        setDisplayedXBlocksCategories: function() {
+            this.parentInfo.category = XBlockUtils.getXBlockType(
+                this.parentInfo.parent.get('category'),
+                // TODO! improve `this.visitedAncestors.length - 2` mysterious calculation
+                this.visitedAncestors[this.visitedAncestors.length - 2]
+            );
+            this.childrenInfo.category = this.categoryRelationMap[this.parentInfo.category];
+        },
+
+        getCurrentLocationIndex: function() {
+            var category, ancestorXBlock, currentLocationIndex;
+
+            if (this.childrenInfo.category === 'component' || this.childrenInfo.children.length === 0) {
+                return currentLocationIndex;
+            }
+
+            category = this.childrenInfo.children[0].get('category');
+            ancestorXBlock = _.find(
+                this.ancestorInfo.ancestors, function(ancestor) { return ancestor.category === category; }
+            );
+
+            if (ancestorXBlock) {
+                _.each(this.childrenInfo.children, function(xblock, index) {
+                    if (ancestorXBlock.display_name === xblock.get('display_name') &&
+                        ancestorXBlock.id === xblock.get('id')) {
+                        currentLocationIndex = index;
+                    }
+                });
+            }
+
+            return currentLocationIndex;
+        },
+
+        getCategoryText: function() {
+            return this.categoriesText[this.childrenInfo.category];
+        },
+
+        getForwardButtonSRText: function() {
+            return StringUtils.interpolate(
+                gettext('Press button to see {XBlockCategory} childs'),
+                {XBlockCategory: this.childrenInfo.category}
+            );
+        },
+
+        getNoChildText: function() {
+            return StringUtils.interpolate(
+                gettext('This {parentCategory} has no {childCategory}'),
+                {
+                    parentCategory: this.parentInfo.category,
+                    childCategory: this.categoriesText[this.childrenInfo.category].toLowerCase()
```

Comment:
What if there is no `this.childrenInfo.category` in `this.categoriesText` map ?  may be in the case of content experiment ?